File: src/main/java/com/devsuatt/todoApp/dto/CreateUserRequestDto.java
package com.devsuatt.todoApp.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;

@AllArgsConstructor
@Getter
public class CreateUserRequestDto {
    @NotBlank(message = "username can not be blank!")
    private String username;
    @NotBlank(message = "password can not be blank!")
    @Size(min = 4, message = "password length must be greater than 3")
    private String password;

}

File: src/main/java/com/devsuatt/todoApp/dto/CreateUserRequestDto.kt
package com.devsuatt.todoApp.dto

import org.hibernate.validator.constraints.UniqueElements
import javax.validation.constraints.NotBlank
import javax.validation.constraints.Size

data class CreateUserRequestDto(
    @field:NotBlank(message = "username can not be null!")
    @field:UniqueElements
    val username: String?,
    @field:NotBlank(message = "password can not be null!")
    @field:Size(min = 4, message = "password length must be greater than 3")
    val password: String?,
)
